File: benchmarks/pandas/bench_cat_to_ordinal.py
"""Benchmark: catToOrdinal on 100k-element categorical Series"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
cats = ["low", "med", "high"]
data = [cats[i % 3] for i in range(ROWS)]
s = pd.Series(pd.Categorical(data))

for _ in range(WARMUP):
    s.astype(pd.CategoricalDtype(categories=cats, ordered=True))

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.astype(pd.CategoricalDtype(categories=cats, ordered=True))
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "cat_to_ordinal", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_cat_to_ordinal.ts
/**
 * Benchmark: catToOrdinal on 100k-element categorical Series
 */
import { Series, catToOrdinal } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const cats = ["low", "med", "high"];
const data = Array.from({ length: ROWS }, (_, i) => cats[i % 3]);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) catToOrdinal(s, cats);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) catToOrdinal(s, cats);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "cat_to_ordinal",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
